### ISSUE TITLE
Mech Problems Refactor Part 3: The Unlogging: Revengence

### ISSUE THREAD

**Issue by necromanceranne** [tgstation/tgstation#54000]
**Mechs (and all sealed vehicles, like the clown car) are completely unlogged when they explode**
Mechs exploding is entirely a consequence of them being turned into sealed vehicles. But as a result, when a mech detonates in an explosion that will easily kill people near it, the explosion is entirely unlogged.

You can chain mech explosions for a fairly sizable boom that is entirely unlogged. Very good.

**Mech guns are entirely unlogged when shot.**
I believe old mechs used to have logging for this.

**Mechs seem to runtime a fuckton when they're destroyed**
This gets spammed a lot every time a mech is destroyed.

```[2020-09-27 08:32:10.089] runtime error: Cannot execute null.return pressure().
 - proc name: process (/obj/vehicle/sealed/mecha/process)
 -   source file: _mecha.dm,421
 -   usr: null
 -   src: the Durand (/obj/vehicle/sealed/mecha/combat/durand)
 -   src.loc: Reinforced glass floor (134,134,2) (/turf/open/transparent/glass/reinforced)
 -   call stack:
 - the Durand (/obj/vehicle/sealed/mecha/combat/durand): process(2)
 - the Durand (/obj/vehicle/sealed/mecha/combat/durand): process(2)
 - Objects (/datum/controller/subsystem/processing/obj): fire(0)
 - Objects (/datum/controller/subsystem/processing/obj): ignite(0)
 - Master (/datum/controller/master): RunQueue()
 - Master (/datum/controller/master): Loop()
 - Master (/datum/controller/master): StartProcessing(0)
```

**Mechs can have their parts deleted from explosions**
Mechs that get hit with explosions seem to have the potential to have their parts destroyed. This isn't the normal interaction where a mechs equipment is destroyed. It is just ex_act deleting their cells and stuff and throwing errors.

```[2020-09-27 08:34:15.818] runtime error: Cannot execute null.return pressure().
 - proc name: get stats part (/obj/vehicle/sealed/mecha/proc/get_stats_part)
 -   source file: mecha_topic.dm,64
```

**Clowncar seems to runtime a lot when it releases it's foam**
This could also be mech related somehow, I'm unsure.

```[2020-09-27 08:47:17.645] runtime error: Cannot execute null.reachableAdjacentTurfs().
 - proc name: spread foam (/obj/effect/particle_effect/foam/proc/spread_foam)
 -   source file: effects_foam.dm,175

[2020-09-27 08:47:18.069] runtime error: Cannot execute null.reachableAdjacentTurfs().
 - Warning: further proc crash messages are being suppressed to prevent overload...
 - 
```

**BONUS ROUND: Wormhole Generators stopped working entirely**
Basically, wormholes? Forghedaboudit.

Helpies @TiviPlus I'm in maint and the badman killed me.

**Comment by TiviPlus** [comment]
Mechs (and all sealed vehicles, like the clown car) are completely unlogged when they explode
unrelated to refactor but will fix anyway

chassis.log_message("Fired from [src.name], targeting [target].", LOG_MECHA)

logging is done

Mechs can have their parts deleted from explosions
was a bug prior but will fix

Clowncar seems to runtime a lot when it releases it's foam
unrelated

Mechs seem to runtime a fuckton when they're destroyed
tested on local:

deleting mechs did not runtime
hitting mechs to destroy them did not runtime
exploding mechs to destroy them did not runtime

error seems to be due to mechs sticking around in processing after being deleted despite
/obj/vehicle/sealed/mecha/Initialize()
	. = ..()
         *snip*
	START_PROCESSING(SSobj, src)


/obj/vehicle/sealed/mecha/Destroy()
         *snip*
	STOP_PROCESSING(SSobj, src)

Runtime would likely be unrelated to mechs
